document/js-api/index.bs

```diff
@@ -0,0 +1,840 @@
+    1. Let |initialSize| be the length of the Table instance's \[[Values]] internal slot.
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |result| be [=grow_table=](|store|, |tableaddr|, |d|).
+    1. If |result| is [=error=], throw a {{RangeError}} exception.
+
+        Note: The above exception may happen due to either insufficient memory or an invalid size parameter.
+
+    6. Set the current agent cluster's [=associated store=] to |store|.
+    1. [=Append=] null to the Table instance's \[[Values]] internal slot |d| &minus; |initialSize| times.
+    1. Return |initialSize|.
+</div>
+
+<div algorithm>
+  The getter of the <dfn attribute for="Table">length</dfn> attribute of {{Table}} returns the length of the table's \[[Values]] internal slot.
+</div>
+
+<div algorithm>
+    The <dfn method for="Table">get(index)</dfn> method, when invoked, performs the following steps:
+    1. Let |values| be the Table instance's \[[Values]] internal slot.
+    1. Let |size| be the length of |values|.
+    1. If |index| &ge; |size|, throw a {{RangeError}} exception.
+    1. Return |values|[|index|].
+</div>
+
+<div algorithm>
+    The <dfn method for="Table">set(index, value)</dfn> method, when invoked, performs the following steps:
+    1. Let |tableaddr| be the Table instance's \[[Table]] internal slot.
+    1. Let |values| be the Table instance's \[[Values]] internal slot.
+    1. If |value| is null, let |elem| be an empty [=function element=].
+    1. Otherwise,
+        1. If |value| does not have a \[[FunctionAddress]] internal slot, throw a {{TypeError}} exception.
+        1. Let |funcaddr| be |value|.\[[FunctionAddress]].
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |store| be [=write_table=](|store|, |tableaddr|, |index|, |funcaddr|).
+    1. If |store| is [=error=], throw a {{RangeError}} exception.
+    1. Set the current agent cluster's [=associated store=] to |store|.
+    1. Set |values|[|index|] to |value|.
+    1. Return undefined.
+</div>
+
+<h3 id="exported-function-exotic-objects">Exported Functions</h3>
+
+A WebAssembly function is made available in JavaScript as an <dfn>Exported Function</dfn>.
+Exported Functions are [=Built-in Function Objects=] which are not constructors, and which have a \[[FunctionAddress]] internal slot.
+This slot holds a [=function address=] relative to the current agent cluster's [=associated store=].
+
+<div algorithm>
+  The <dfn>name of the WebAssembly function</dfn> |funcaddr| is found by performing the following steps:
+
+    1. Let |store| be the current agent cluster's [=associated store=].
+    1. Let |funcinst| be |store|.𝖿𝗎𝗇𝖼𝗌[|funcaddr|].
+    1. If |funcinst| is of the form {𝗍𝗒𝗉𝖾 |functype|, 𝗁𝗈𝗌𝗍𝖼𝗈𝖽𝖾 |hostfunc|},
+      1. Assert: |hostfunc| is a JavaScript object and [=IsCallable=](|hostfunc|) is true.
+      1. Let |name| be ? [=Get=](|hostfunc|, "name").
+      1. Return ? [=ToString=](|name|).
```

Comment:
In Firefox and Chrome (testing on Linux just now; probably the others too), there is no special case here for host functions and in all cases the function index is used as the name of an exported wasm function.

Reply:
I can't remember why I thought at the time that it would be reasonable to specify this behavior. I'll remove this case.

Reply:
Their function index is the function index of the function import that was re-exported.